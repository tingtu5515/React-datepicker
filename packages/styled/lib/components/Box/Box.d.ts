import {
  BottomProps,
  GridAreaProps,
  HeightProps,
  LeftProps,
  PositionProps,
  RightProps,
  SpaceProps,
  TopProps,
  WidthProps,
} from 'styled-system'
interface BoxProps
  extends GridAreaProps,
    HeightProps,
    SpaceProps,
    PositionProps,
    TopProps,
    BottomProps,
    LeftProps,
    RightProps,
    WidthProps {}
declare const Box: import('styled-components').StyledComponent<'div', any, BoxProps, never>
export default Box
