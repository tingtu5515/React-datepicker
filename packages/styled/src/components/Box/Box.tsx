import {
  bottom,
  BottomProps,
  compose,
  gridArea,
  GridAreaProps,
  height,
  HeightProps,
  left,
  LeftProps,
  position,
  PositionProps,
  right,
  RightProps,
  space,
  SpaceProps,
  top,
  TopProps,
  width,
  WidthProps,
  zIndex,
  ZIndexProps,
} from 'styled-system'
import styled from 'styled-components'

interface BoxProps
  extends GridAreaProps,
    HeightProps,
    SpaceProps,
    PositionProps,
    TopProps,
    BottomProps,
    LeftProps,
    RightProps,
    ZIndexProps,
    WidthProps {}
const composeBoxStyles = compose(
  gridArea,
  height,
  space,
  width,
  position,
  top,
  left,
  right,
  bottom,
  zIndex,
)

const Box = styled('div')<BoxProps>`
  box-sizing: border-box;
  ${composeBoxStyles}
`

export default Box
