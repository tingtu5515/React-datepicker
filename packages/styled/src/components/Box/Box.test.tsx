import React from 'react'
import {render} from 'react-testing-library'
import Box from '.'

test('should match snapshot', () => {
  const {container} = render(
    <Box
      gridArea="test"
      height="150px"
      width="150px"
      p="10px"
      position="relative"
      top="10px"
      left="15px"
      right="20px"
      bottom="30px"
      zIndex={1}
    >
      <div>test1</div>
      <div>test2</div>
      <div>test3</div>
      <div>test4</div>
      <div>test5</div>
    </Box>,
  )
  expect(container).toMatchSnapshot()
})
