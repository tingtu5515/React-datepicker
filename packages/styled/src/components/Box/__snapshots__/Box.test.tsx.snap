// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  grid-area: test;
  height: 150px;
  width: 150px;
  padding: 10px;
  position: relative;
  top: 10px;
  left: 15px;
  right: 20px;
  bottom: 30px;
  z-index: 1;
}

<div>
  <div
    class="c0"
    height="150px"
    width="150px"
  >
    <div>
      test1
    </div>
    <div>
      test2
    </div>
    <div>
      test3
    </div>
    <div>
      test4
    </div>
    <div>
      test5
    </div>
  </div>
</div>
`;
